src/compiler/checker.ts

```diff
@@ -22983,6 +22983,23 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
         return getCachedType(key) ?? setCachedType(key, mapType(type, getBaseTypeOfLiteralType));
     }
 
+    // This is the same as getBaseTypeOfLiteralType, but checks EnumLiteral last so we get
+    // a specific literal's base type rather than the enum's base type.
+    function getBaseTypeOfLiteralTypeForComparison(type: Type): Type {

```

Comment:
FWIW I tried changing `getBaseTypeOfLiteralType` and that of course failed because it's used in places where you do actually want to go to the enum type itself.

Reply:
Another option here is to add an optional parameter to getBaseTypeOfLiteralType and fiddle with the internals. Happy to do whatever.